src/novel_applications/defi/clob.md

```diff
@@ -0,0 +1,167 @@
+
+In crypto markets, the dominant venue for exchange has been centralized order book-based exchanges (CEX). There is high demand for on-chain order-book exchanges; see [here](https://twitter.com/EliBenSasson/status/1614575237122629633?s=20) 40 TPS only on dYdX. Onchain exchanges are mostly AMM-based exchanges that cannot provide quite the same benefits: 
+
+Order-book exchanges:
+- Provide price efficiency
+- Reduce the risk of slippage and have no impermanent loss
+- Are widely accepted by both institutional and retail traders alike 
+- Are easier to use
+
+To run an on-chain order book, several features are needed:
+
+Needed:
+1. High TPS - builders ask for 30-40 and some even more
+2. Low finality - sub-second finality would be optimal
+3. Quick and cheap order updates - Order updates should be almost instantaneous and free, especially for Market Makers
+
+Nice to have:
+4. Privacy at the base layer - to not reveal a trading strategy 
+5. Matching engine - FIFO/Pro-Rata should also run on-chain to ensure "best execution"
+6. MEV protection - to ensure "best execution"
+
+(Are there more required features? Do we need to cancel the order quickly?)
+
+## Miden: An Improved Execution Infrastructure
+
+Order-book exchanges on account-based blockchains are hard to implement. The algorithms of matching engines, commonly implemented in centralized exchanges (CEXs) based on the CLOB model, consume a large number of resources, making it challenging to deploy and run on the Ethereum blockchain, see [here](https://www.gdx.org/gridex-whitepaper.pdf). Some DEXs have attempted to solve this issue by implementing matching engines and order books off-chain - dYdX (on StarkEX) or 0x. However, such solutions increase the risk of centralization (to a certain extent) and limit their ability to interact with other protocols, making it impossible to utilize or contribute to the open ecosystem of Ethereum effectively.
+
+Polygon Miden offers the required features (1-6) while simultaneously, computational integrity is secured by Ethereum, and liquidity is provided by Ethereum and the Polygon ecosystem. 
+
+Polygon Miden is designed to have high TPS sustainably. Orders can be updated for free and in no time using updateable transactions. Polygon Miden offers different levels of privacy at the base layer. Matching engines can run off-chain, but there can be a zk-proof of correct execution to be verified by the network. 
+    
+For finality, we need to define what we mean by that. 
+
+## A Powerful Order Book on Miden
+    
+Let's collect ideas for implementing a private on-chain order book exchange on Miden. Given the [Miden Architecture](https://0xpolygonmiden.github.io/miden-base/architecture.html), we can design a simple onchain order book exchange. 
```

Comment:
This link doesn't work anymore, I think you need to link to the new docs https://docs.polygon.technology/miden/miden-base/architecture/overview/